### PR TITLE
aa_autodep.pm: delete file only if it exists

### PR DESCRIPTION
**Pull request by paolostivanin**
Ticket: https://progress.opensuse.org/issues/138392
VRs:
* 12-SP5: https://openqa.suse.de/tests/12634227
* 15-SP1: https://openqa.suse.de/tests/12634215
* 15-SP2: https://openqa.suse.de/tests/12634216
* 15-SP4: https://openqa.suse.de/tests/12634217

### CODE REVIEW

File: tests/security/apparmor/aa_autodep.pm [hunk 1]
```diff
@@ -74,7 +74,7 @@ sub run {
 
     # delete cache file of aa-autodep generated profile, so that the next reload creates a fresh cache of /etc/apparmor.d/usr.sbin.nscd
     # (wouldn't happen without deleting the cache file because the cache timestamp is newer than the profile (+ used abstractions) timestamp)
-    assert_script_run "rm -v /var/cache/apparmor/*/usr.sbin.nscd";
+    assert_script_run "rfind /var/cache/apparmor/ -name usr.sbin.nscd -delete";
```
Comment (jknphy) [inline tests/security/apparmor/aa_autodep.pm:77]: rfind? typo?

Comment (paolostivanin) [reply tests/security/apparmor/aa_autodep.pm:77]: indeed it is xD

File: tests/security/apparmor/aa_autodep.pm [hunk 2]
```diff
@@ -74,7 +74,8 @@ sub run {
 
     # delete cache file of aa-autodep generated profile, so that the next reload creates a fresh cache of /etc/apparmor.d/usr.sbin.nscd
     # (wouldn't happen without deleting the cache file because the cache timestamp is newer than the profile (+ used abstractions) timestamp)
-    assert_script_run "rm -v /var/cache/apparmor/*/usr.sbin.nscd";
+    my $target_dir = (is_sle('>=15') || is_opensuse) ? '/var/cache/apparmor/' : '/var/lib/apparmor/cache/';
```
Comment (jknphy) [inline tests/security/apparmor/aa_autodep.pm:77]: but in sle-15-sp1 didn't find this path...

Comment (paolostivanin) [reply tests/security/apparmor/aa_autodep.pm]: sleep deprivation is kicking in xD thanks for noticing, fixed

**Pull request merged**